src/dscanner/stats.d

```diff
@@ -30,25 +32,122 @@ pure nothrow bool isLineOfCode(IdType t)
 	}
 }
 
-ulong printTokenCount(Tokens)(File output, string fileName, ref Tokens tokens)
+pure nothrow bool __isLineOfCode(TOK t)
 {
+	switch (t)
+	{
+	case TOK.semicolon:
+	case TOK.do_:
+	case TOK.while_:
+	case TOK.if_:
+	case TOK.else_:
+	case TOK.for_:
+	case TOK.foreach_:
+	case TOK.foreach_reverse_:
+	case TOK.switch_:
+	case TOK.case_:
+	case TOK.default_:
+		return true;
+	default:
+		return false;
+	}
+}
+
+ulong printTokenCount(File output, string fileName, ref Lexer lexer)
+{
+	TOK lastTOK = TOK.reserved;
 	ulong c;
-	foreach (ref t; tokens)
+	foreach (ref t; lexer)
 	{
-		c++;
+		if (lastTOK == TOK.whitespace && t == TOK.whitespace)
```

Comment:
It seems that dscanner is able to configure libdparse to count whitespaces or not. We should probably add this configuration option to dmd as well, that way we make sure that we improve the library interface and keep the changes in dscanner to a minimum.